src/javascript/crypto/e2e/openpgp/pgpmime/types.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @license
+ * Copyright 2015 Google Inc. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Type definitions for MIME modules
+ * @author yzhu@yahoo-inc.com (Yan Zhu)
+ */
+
+goog.provide('e2e.openpgp.pgpmime.types.Attachment');
+goog.provide('e2e.openpgp.pgpmime.types.ContentAndHeaders');
+goog.provide('e2e.openpgp.pgpmime.types.Entity');
+goog.provide('e2e.openpgp.pgpmime.types.Header');
+goog.provide('e2e.openpgp.pgpmime.types.HeaderValue');
+goog.provide('e2e.openpgp.pgpmime.types.MailContent');
+
+
+/**
+ * @typedef {{body: string, subject: (string|undefined),
```

Comment:
Shouldn't this include attachments? I'd make MailContent a submessage of this message type.